src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -208,6 +292,82 @@ public static bool TryParseInt64(string value, out long result)
             return long.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
         }
 
+        internal static unsafe bool TryParseInt32(StringSegment value, out int result)
```

Comment:
Does the old one need to parse negatives? I'd be surprised. You can probably have the old one call the new one.

Add some overflow checks.